_posts/2023-03-06-opentelemetry.md

```diff
@@ -0,0 +1,245 @@
+
+This blog post will show you, how to use OpenTelemetry within the Strimzi project and what are the differences with OpenTracing which is now deprecated and will be removed in the coming months.
+
+## What is OpenTelemetry?
+
+OpenTelemetry is an open source and vendor neutral project providing tools, APIs, and SDKs used to generate, collect and export telemetry data.
+It is part of the Cloud Native Computing Foundation and it is available across several languages.
+It is not just about tracing but it has support for metrics and logs as well.
+
+> The Strimzi project supports with OpenTelemetry is about tracing only. There are no actual plans to support OpenTelemetry metrics and logs at this time.
+
+While the OpenTracing support in Strimzi is strictly tight to the Jaeger implementation, by using the Jaeger client underneath, with OpenTelemetry we have a different approach.
+The tracer uses an exporter in order to export traces in a specific format to the backend system.
+While a Jaeger based exporter is available, to allow current users to continue using the Jaeger protocol, the OTLP (OpenTeLemetry Protocol) one is used by default and the OpenTelemetry community encourage to move to it.
+You can also use Zipkin or developing your own exporter for a custom backend system.
+
+Using a specific exporter protocol means that the corresponding backend system has to support it.
+It has to provide an endpoint to which the traces are sent by using that specific protocol.
+For example, you can still use Jaeger backend with the OTLP exporter protocol, because newer Jaeger releases since 1.35 exposes an OTLP endpoint.
+You could also decide to use the usual Jaeger protocol endpoint by setting the corresponding Jaeger exporter in your system.
+
+From an application perspective, the way OpenTelemetry works is similar to OpenTracing.
+The tracer is part of your application and it provides the API for creating trace spans and using the configured exporter to send them.
+Focusing on Apache Kafka, the application could be a producer or consumer, or one of the components managed by the Strimzi operator, such as Kafka Connect, Kafka Mirror Maker(s) and the HTTP bridge.
+
+As it happens with OpenTracing, even the OpenTelemetry exporter doesn't send all traces but often just part of them by using sampling.
+The sampler mechanism can be configured through environment variables.
+
+## How to enable OpenTelemetry
+
+Before adding the support for OpenTelemetry, it was already possible to enable the distributed tracing on the `KafkaConnect`, `KafkaMirrorMaker`, `KafkaMirrorMaker2` and `KafkaBridge` custom resources by setting the `spec.tracing.type` property to `jaeger` to use OpenTracing.
+Since the Strimzi 0.32.0 release, you can set the corresponding `opentelemetry` value for using OpenTelemetry instead.
```

Comment:
Doesn't 0.32 still use Jaeger instead of OLTP? Wouldn't it be better for the sake of the post to just pretend its 0.33+ as you cover only OLTP?

Reply:
Yeah you are right, I will fix to 0.33.0 there.